src/toxcore/dht/dht_node.rs

```diff
@@ -64,6 +64,20 @@ impl<T> SockAndTime<T> {
     pub fn is_ping_interval_passed(&self) -> bool {
         self.last_ping_req_time.map_or(true, |time| clock_elapsed(time) >= Duration::from_secs(PING_INTERVAL))
     }
+
+    /// Get address if it should be pinged and update `last_ping_req_time`.
+    pub fn ping_addr(&mut self) -> Option<T> {
+        if let Some(saddr) = self.saddr {
```

Comment:
Nope, this will return `Some` in all cases when `self.saddr` is some. I need `None` if the condition `!self.is_discarded() && self.is_ping_interval_passed()` is not satisfied.